mirascope/base/extractors.py

```diff
@@ -89,12 +90,30 @@ async def extract_async(self, retries: int = 0) -> ExtractedTypeT:
     #     ...  # pragma: no cover
 
     ############################## PRIVATE METHODS ###################################
+    def _retry(
```

Comment:
I don't understand the point of this function being separate. Ultimately this will require updating it everywhere vs. just updating the `_extract` method to handle retries internally so every provider automatically works and we only have to make changes in the base extractor?

This is specific to `_extract` anyway, and `_extract_async`, `_stream`, and `_stream_async` will require their own specific handling anyway, so seems best to just embed this retry functionality into each of those functions as a replacement of our current recursive method?